junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/ExtensionValuesStore.java

```diff
@@ -110,7 +110,16 @@ Object remove(Namespace namespace, Object key) {
 
 	private StoredValue getStoredValue(Namespace namespace, Object key) {
 		CompositeKey compositeKey = new CompositeKey(namespace, key);
-		return this.storedValues.get(compositeKey);
+
+		/*
+		 * This method can be called from another ExtensionValuesStore instance,
+		 * so we must synchronized on the monitor.
```

Comment:
> It can be, but is there currently anywhere
> In the code 

Yes. From `getOrComputeIfAbsent()` where the code accesses the parent store's map without obtained the parent's lock.